FirebaseAnalyticsSwift/Tests/SwiftUnit/AnalyticsAPITests.swift

```diff
@@ -0,0 +1,170 @@
+//
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+// MARK: This file is used to evaluate the experience of using Analytics APIs in Swift.
+
+import FirebaseAnalytics
+import Foundation
+
+final class AnalyticsAPITests {
+  func usage() {
+    // MARK: - Analytics
+
+    Analytics.logEvent("event_name", parameters: ["param": 1])
+    Analytics.setUserProperty("value", forName: "name")
+    Analytics.setUserID("user_id")
+    Analytics.setAnalyticsCollectionEnabled(true)
+    Analytics.setSessionTimeoutInterval(3600.0)
+    let _: String? = Analytics.appInstanceID()
+    Analytics.resetAnalyticsData()
+    Analytics.setDefaultEventParameters(["default": 100])
+
+    // MARK: - AppDelegate
+
+    Analytics.handleEvents(forBackgroundURLSession: "session_id", completionHandler: {})
```

Comment:
Please add async/await variation of this API. See https://github.com/firebase/firebase-ios-sdk/blob/master/FirebaseDatabase/Tests/Unit/Swift/DatabaseAPITests.swift#L141

Reply:
Thanks for the info! However, it will take some effort (design/review/implement) to add a new API. If you don't mind, I want to merge this pull-request first.

Reply:
Done, thanks.